### PR TITLE
[Sqlparser] VALUES(...) should take a fully-qualified column name

### PR DESCRIPTION
**Pull request by danieltahara**
A values expression is allowed to take a fully qualified (table.col), but the current parser only allows unqualified columns. I have modified the grammar, plan builder, and AST accordingly.

Signed-off-by: Daniel Tahara <tahara@dropbox.com>

### CODE REVIEW

**Comment by sougou** [comment]
More changes are needed. See the build failure :).

**Comment by danieltahara** [comment]
Yep yep :) It's easier to punt these up here and have all the ci run than set it up locally :P

**Comment by danieltahara** [comment]
Added a negative test (mistmatched table name) and rebased.

**Comment by acharis** [comment]
LGTM

**Comment by danieltahara** [comment]
bump

**Comment by danieltahara** [comment]
Bump :)

**Comment by sougou** [comment]
The ETA on this is probably weekend. I'll try to find time before if possible.

**Comment by danieltahara** [comment]
bump :)

**Comment by sougou** [comment]
I've started working my way through. I should get to this soon enough now.

File: go/vt/vttablet/tabletserver/planbuilder/dml.go [hunk 1]
```diff
@@ -432,7 +432,13 @@ func analyzeInsertNoType(ins *sqlparser.Insert, plan *Plan, table *schema.Table)
 	var formatErr error
 	plan.UpsertQuery = GenerateUpdateOuterQuery(upd, tableAlias, func(buf *sqlparser.TrackedBuffer, node sqlparser.SQLNode) {
 		if node, ok := node.(*sqlparser.ValuesFuncExpr); ok {
-			colnum := ins.Columns.FindColumn(node.Name)
+			if !node.Name.Qualifier.IsEmpty() && node.Name.Qualifier != ins.Table {
+				formatErr = vterrors.Errorf(vtrpcpb.Code_NOT_FOUND,
+					"could not find qualified column %v in table %v",
+					node.Name, ins.Table)
```
[the remark this answers is not in the document]
Comment (danieltahara) [reply go/vt/vttablet/tabletserver/planbuilder/dml.go]: done, and updated tests.

**Comment by danieltahara** [comment]
fixed. :)

**Pull request merged**